python/hsfs/usage.py

```diff
@@ -0,0 +1,255 @@
+        self._user_id = None
+        self._backend_version = None
+        self._timezone = None
+
+    def _get_lib_version(self, lib_name):
+        try:
+            lib = __import__(lib_name)
+            return lib.__version__
+        except ImportError:
+            return ""
+
+    def get_hsml_version(self):
+        if self._hsml_version is None:
+            self._hsml_version = self._get_lib_version("hsml")
+        return self._hsml_version
+
+    def get_hsfs_version(self):
+        if self._hsfs_version is None:
+            self._hsfs_version = self._get_lib_version("hsfs")
+        return self._hsfs_version
+
+    def get_hopsworks_version(self):
+        if self._hopsworks_version is None:
+            self._hopsworks_version = self._get_lib_version("hopsworks")
+        return self._hopsworks_version
+
+    def get_python_version(self):
+        if not self._python_version:
+            self._python_version = platform.python_version()
+        return self._python_version
+
+    def get_user_id(self):
+        if not self._user_id:
+            hopsworks_dir = _create_hopsworks_dir_if_not_exist()
+            user_id_file = join(hopsworks_dir, _USER_ID_FILE)
+            if os.path.exists(user_id_file):
+                with open(user_id_file, "r") as fr:
+                    self._user_id = fr.read().rstrip()
+            else:
+                with open(user_id_file, "w") as fw:
+                    self._user_id = _generate_user_id()
+                    fw.write(self._user_id)
+        return self._user_id
+
+    def get_platform(self):
+        if not self._platform:
+            self._platform = platform.platform()
+        return self._platform
+
+    def get_backend_host_name(self):
+        return _backend_hostname
+
+    def get_backend_version(self):
+        return _backend_version
+
+    def get_timezone(self):
+        if self._timezone is None:
+            self._timezone = datetime.now().astimezone().tzinfo
+        return self._timezone
+
+
+class MethodCounter:
+    def __init__(self):
+        self.method_counts = {}
+        random.seed(42)
+
+    def add(self, m):
+        s = self._get_method_name(m)
+        self.method_counts[s] = self.method_counts.get(s, 0) + 1
+
+    def get_count(self, m):
+        s = self._get_method_name(m)
+        return self.method_counts.get(s, 0)
+
+    def should_sample(self, m):
+        cnt = self.get_count(m)
+        if cnt < 100:
+            return True
+        elif cnt < 1000:
+            return random.random() < 0.1
+        elif cnt < 10000:
+            return random.random() < 0.01
+        else:
+            return random.random() < 0.001
+
+    def _get_method_name(self, m):
+        return m.__module__ + m.__name__
+
+
+logging.basicConfig(stream=sys.__stdout__)
+_logger = logging.getLogger(__name__)
+_logger.setLevel(logging.INFO)
+_conn = http.client.HTTPSConnection("usage.hops.works")
+_executor = concurrent.futures.ThreadPoolExecutor(max_workers=2)
```

Comment:
Can we make `max_workers` configurable as well? Maybe through another environment variable?